### PR TITLE
[4004] Integrate item selector into dialogs 

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4004

### CODE REVIEW

**Comment by rart** [comment]
Please integrate with New content dialog, Dependencies dialog, History & View Versions too.

File: ui/app/src/state/reducers/items.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export const fetchPath = createAction<string>('FETCH_PATH');
+export const fetchPathComplete = createAction<GetChildrenResponse>('FETCH_PATH_COMPLETE');
+export const fetchPathFailed = createAction<any>('FETCH_PATH_COMPLETE_FAILED');
+
+const initialState: ItemsStateProps = {
+  byId: null,
+  isFetching: null,
+  error: null,
+  items: null,
+  path: 'site/website'
```
Comment (rart) [inline ui/app/src/state/reducers/items.ts]: Path should be at widget state. This should just be an item container.

File: ui/app/src/state/reducers/items.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  {
+    [fetchPath.type]: (state, { payload }) => ({
+      ...state,
+      ...payload,
+      isFetching: true
+    }),
+    [fetchPathComplete.type]: (state, { payload }) => ({
+      ...state,
+      byId: {...state.byId, ...createLookupTable(payload)},
+      items: payload.map(item => item.id),
```
Comment (rart) [inline ui/app/src/state/reducers/items.ts]: Belongs to widget.

File: ui/app/src/models/Item.ts [hunk 3]
```diff
@@ -89,3 +92,11 @@ export interface LegacyItem {
   children: LegacyItem[];
   [prop: string]: any;
 }
+
+export interface ItemsStateProps {
+  byId: LookupTable<SandboxItem>,
+  error: ApiResponse;
+  isFetching: boolean;
+  items: string[],
+  path: string
```
Comment (rart) [inline ui/app/src/models/Item.ts]: Items and path belongs to widgets

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/models/Item.ts [hunk 4]
```diff
@@ -89,3 +93,21 @@ export interface LegacyItem {
   children: LegacyItem[];
   [prop: string]: any;
 }
+
+export interface Consumer extends PaginationOptions {
```
Comment (rart) [inline ui/app/src/models/Item.ts]: Move this out

File: ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx [hunk 5]
```diff
@@ -14,142 +14,166 @@
...
-          variant={titleVariant || 'body1'}
-          className={clsx(classes.title, propClasses?.title)}
-        >
-          {label}
-        </Typography>
-        <LabelIcon className={clsx(classes.labelIcon, propClasses?.labelIcon)} />
-        <Typography variant={labelVariant || 'body1'}>{selectItem.label}</Typography>
-      </div>
+      <Typography
+        variant={titleVariant || 'body1'}
```
Comment (rart) [inline ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx]: set default on the prop

File: ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx [hunk 6]
```diff
@@ -14,142 +14,166 @@
...
+        className={clsx(classes.title, propClasses?.title)}
+      >
+        {label}
+      </Typography>
+      {
+        selectedItem &&
+        <div className={classes.selectedItem}>
+          <ItemIcon className={clsx(classes.itemIcon, propClasses?.itemIcon)} />
+          <Typography variant={labelVariant || 'body1'}>{selectedItem.label}</Typography>
```
Comment (rart) [inline ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx]: default on the prop

File: ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx [hunk 7]
```diff
@@ -14,142 +14,166 @@
...
+          horizontal: 'right'
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right'
+        }}
+      >
+        <Breadcrumbs
+          keyword={consumer?.keywords}
+          breadcrumb={consumer ? consumer?.breadcrumb : []}
```
Comment (rart) [inline ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx]: consumer?.breadcrumb ?? []

File: ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx [hunk 8]
```diff
@@ -14,142 +14,166 @@
...
+          onSearch={() => {
+          }}
+          onCrumbSelected={onBreadcrumbSelected}
+        />
+        <SuspenseWithEmptyState resource={itemsResource}>
+          <PathNavigatorList
+            leafs={consumer ? consumer?.leafs : []}
```
Comment (rart) [inline ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx]: consumer?.leafs ?? []

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/ComponentMenu.tsx [hunk 9]
```diff
@@ -117,9 +117,8 @@ export default function ComponentMenu(props: ComponentMenuProps) {
         break;
       }
       case 'history': {
-        dispatch(fetchItemVersions({
-          path: models[modelId].craftercms.path || models[parentId].craftercms.path
-        }));
+        const path = models[modelId].craftercms.path || models[parentId].craftercms.path;
+        dispatch(fetchItemVersions({ path }));
         dispatch(showHistoryDialog());
```
Comment (rart) [inline ui/app/src/components/ComponentMenu.tsx]: Use batchActions where ever required

File: ui/app/src/modules/Content/History/CompareVersionsDialog.tsx [hunk 10]
```diff
@@ -95,35 +113,38 @@ export default function CompareVersionsDialog(props: CompareVersionsDialogProps)
...
+      compareVersionsBranch,
+      contentTypesBranch
+    },
+    {
+      shouldResolve: ({ compareVersionsBranch, contentTypesBranch }) => (
+        Boolean(
+          compareVersionsBranch.compareVersions &&
+          contentTypesBranch.byId &&
+          !compareVersionsBranch.isFetching &&
+          !contentTypesBranch.isFetching
```
Comment (rart) [inline ui/app/src/modules/Content/History/CompareVersionsDialog.tsx]: With all the &&, the boolean constructor doesn’t make sense

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/modules/Content/Dependencies/DependenciesDialog.tsx [hunk 11]
```diff
@@ -343,12 +346,38 @@ function DependenciesDialogUI(props: DependenciesDialogUIProps) {
...
+          {/*      dispatch(fetchItemVersions({ path: item.path }));*/}
+          {/*    } else {*/}
+          {/*      dispatch(fetchChildrenByPath({ id: dependenciesDialogID, path: item.path }));*/}
+          {/*    }*/}
+          {/*  }}*/}
+          {/*  onItemClicked={(item) => {*/}
+          {/*    setOpenSelector(false);*/}
+          {/*    dispatch(fetchItemVersions({ path: item.path }));*/}
+          {/*  }}*/}
+          {/*/>*/}
```
Comment (rart) [inline ui/app/src/modules/Content/Dependencies/DependenciesDialog.tsx]: x

File: ui/app/src/services/content.ts [hunk 12]
```diff
@@ -77,6 +77,12 @@ export function getLegacyItem(site: string, path: string): Observable<LegacyItem
   );
 }
 
+export function getItem(site: string, path: string): Observable<SandboxItem> {
```
Comment (rart) [inline ui/app/src/services/content.ts]: Rename to getSandboxItem

File: ui/app/src/utils/hooks.ts [hunk 13]
```diff
@@ -184,22 +184,23 @@ export function useStateResource<ReturnType = unknown, SourceType = unknown, Err
...
   const [resource, resolve, reject] = bundle;
-  const effectFn = () => {
-    const { shouldRenew, shouldReject, shouldResolve, errorSelector, resultSelector } = checkers;
+  const ref = useRef<any>();
+  ref.current = checkers;
+
+  // Purposely not adding checkers on to the effect dependencies to avoid
+  // consumer re-renders to trigger the effect. `checkers` should be taken
+  // as a "initialValue" sort of param. The functions should not mutate
+  // throughout the component lifecycle.
```
Comment (rart) [inline ui/app/src/utils/hooks.ts]: The effect is now in sync with it's dependency. Remove comment.

File: ui/app/src/utils/path.ts [hunk 14]
```diff
@@ -68,11 +93,27 @@ export function withIndex(path: string): string {
...
+  return splitPath.join('/');
+}
+
+export function getParentsFromPath(path: string): string[] {
+  let splitPath = withoutIndex(path).split('/');
+  splitPath.pop();
+  return splitPath.map( function(value, i) {
+    return '/' + splitPath.slice(1, i + 1).join('/');
+  }).splice(2);
+}
```
Comment (rart) [inline ui/app/src/utils/path.ts]: Format

File: ui/app/src/utils/path.ts [hunk 15]
```diff
@@ -68,11 +93,27 @@ export function withIndex(path: string): string {
...
 export default {
   getPathFromPreviewURL,
   getPreviewURLFromPath,
   getQueryVariable,
   parseQueryString,
+  itemsFromPath,
   withoutIndex,
-  withIndex
+  withIndex,
+  getParentsFromPath
```
Comment (rart) [inline ui/app/src/utils/path.ts:117]: Add all new methods

File: static-assets/components/cstudio-common/common-api.js [hunk 16]
```diff
@@ -721,7 +721,7 @@ var nodeOpen = false,
           type: 'SHOW_DELETE_DIALOG',
           payload: {
             open: true,
-            items,
+            items: items.map(item => CrafterCMSNext.services.content.parseLegacyItemToSandBoxItem(item)),
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Move to utils/content

File: static-assets/components/cstudio-common/common-api.js [hunk 17]
```diff
@@ -748,84 +748,61 @@ var nodeOpen = false,
...
+        })
 
-              this.on('showRequest', function (evt, args) {
-                dialogue.show();
-              });
-            }
-          },
-          true
-        );
+        CrafterCMSNext.system.store.dispatch({ type: 'SHOW_HISTORY_DIALOG' });
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: use batchActions

File: static-assets/components/cstudio-common/common-api.js [hunk 18]
```diff
@@ -748,84 +748,61 @@ var nodeOpen = false,
...
+
+        CrafterCMSNext.system.store.dispatch({
+          type: 'SHOW_HISTORY_DIALOG',
+          payload: {
+            onClose: {
+              type: 'LEGACY_DIALOG_CALLBACK',
+              payload: { id: eventIdOnClose }
+            }
+          }
+        });
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js:810]: batchActions

File: static-assets/components/cstudio-common/common-api.js [hunk 19]
```diff
@@ -748,84 +748,61 @@ var nodeOpen = false,
...
+        CrafterCMSNext.createLegacyCallbackListener(eventIdOnClose, () => {
+          // TODO: we need to found a way to know when the dispatch to Revert finished and it is true
+          // to call an OnSuccess Callback
+          eventNS.data = contentObj;
+          eventNS.typeAction = '';
+          eventNS.oldPath = null;
+          document.dispatchEvent(eventNS);
+          amplify.publish('HISTORY_REVERT');
+          // end of TODO
+          CrafterCMSNext.system.store.dispatch({ type: 'RESET_VERSIONS_STATE' });
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: This was no longer necessary, was it?

File: ui/app/src/components/ComponentMenu.tsx [hunk 20]
```diff
@@ -99,43 +87,41 @@ export default function ComponentMenu(props: ComponentMenuProps) {
...
           scheduling: 'now'
         }));
         break;
       }
       case 'history': {
-        dispatch(fetchItemVersions({
-          path: models[modelId].craftercms.path || models[parentId].craftercms.path
+        dispatch(fetchItemVersions({ item: item }));
+        dispatch(showHistoryDialog({
+          //onRevert
```
Comment (rart) [inline ui/app/src/components/ComponentMenu.tsx]: x

Comment (rart) [reply ui/app/src/components/ComponentMenu.tsx]: batchactions

File: ui/app/src/components/Dialogs/ConfirmDialog.tsx [hunk 21]
```diff
@@ -24,6 +23,8 @@ import Button from '@material-ui/core/Button';
 import React, { PropsWithChildren } from 'react';
 import { defineMessages, useIntl } from 'react-intl';
 import StandardAction from '../../models/StandardAction';
+import { Dialog } from '@material-ui/core';
+import { useDialogOnClosed } from '../../utils/hooks';
```
Comment (rart) [inline ui/app/src/components/Dialogs/ConfirmDialog.tsx]: Let’s rename to useOnUnmount

File: ui/app/src/components/SystemStatus/GlobalDialogManager.tsx [hunk 22]
```diff
@@ -152,7 +161,9 @@ function GlobalDialogManager() {
       <HistoryDialog
         open={state.history.open}
         versionsBranch={versionsBranch}
+        //onRevert={createCallback(state.history.onRevert, dispatch)}
```
Comment (rart) [inline ui/app/src/components/SystemStatus/GlobalDialogManager.tsx]: x

File: ui/app/src/models/Version.ts [hunk 23]
```diff
@@ -14,7 +14,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { LegacyItem } from './Item';
+import { LegacyItem, SandboxItem } from './Item';
```
Comment (rart) [inline ui/app/src/models/Version.ts:17]: Check terminal warnings. Are you still using LegacyItem here for example?

File: ui/app/src/modules/Content/Authoring/NewContentDialog.tsx [hunk 24]
```diff
@@ -44,6 +43,11 @@ import { SuspenseWithEmptyState } from '../../../components/SystemStatus/Suspenc
 import { LegacyFormConfig } from '../../../models/ContentType';
 import { Resource } from '../../../models/Resource';
 import StandardAction from '../../../models/StandardAction';
+import { useDispatch } from 'react-redux';
+import SingleItemSelector from './SingleItemSelector';
+import { SandboxItem } from '../../../models/Item';
+import { showErrorDialog } from '../../../state/reducers/dialogs/error';
+import { Dialog } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/modules/Content/Authoring/NewContentDialog.tsx]: Make sure to import the dialog from its file and not from core (check all over the dialogs where you did the find & replace)

File: ui/app/src/modules/Content/Authoring/NewContentDialog.tsx [hunk 25]
```diff
@@ -187,20 +188,36 @@ function ContentTypesGrid(props: ContentTypesGridProps) {
 }
 
 export default function NewContentDialog(props: NewContentDialogProps) {
+  return (
+    <Dialog
+      open={props.open}
+      onClose={props.onClose}
+
```
Comment (rart) [inline ui/app/src/modules/Content/Authoring/NewContentDialog.tsx]: x empty line

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 26]
```diff
@@ -64,19 +65,21 @@ interface DeleteDialogUIProps {
...
 }
 
 export type DeleteDialogProps = PropsWithChildren<
   DeleteDialogBaseProps & {
-    onClose(): any;
-    onDismiss(): any;
-    onSuccess?(response?: any): any;
-  }
+  onClose(): any;
+  onClosed(): any;
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: needn’t be mandatory

File: ui/app/src/modules/Content/History/CompareVersionsDialog.tsx [hunk 27]
```diff
@@ -70,136 +84,194 @@ interface CompareVersionsDialogProps extends CompareVersionsDialogBaseProps {
...
 }
 
 export default function CompareVersionsDialog(props: CompareVersionsDialogProps) {
-  const { open, rightActions, selectedA, selectedB, onDismiss, onClose, versionsBranch, contentTypesBranch } = props;
-  const { count, page, limit, selected, compareVersionsBranch, current } = versionsBranch;
+  return (
+    <Dialog
+      open={props.open}
+      onClose={props.onClose}
+
```
Comment (rart) [inline ui/app/src/modules/Content/History/CompareVersionsDialog.tsx]: x

File: ui/app/src/modules/Content/History/CompareVersionsDialog.tsx [hunk 28]
```diff
@@ -70,136 +84,194 @@ interface CompareVersionsDialogProps extends CompareVersionsDialogBaseProps {
...
+      fullWidth={true}
```
Comment (rart) [inline ui/app/src/modules/Content/History/CompareVersionsDialog.tsx]: use shortcut syntax

File: ui/app/src/modules/Content/History/ViewVersionDialog.tsx [hunk 29]
```diff
@@ -122,25 +126,41 @@ interface ViewVersionDialogProps extends ViewVersionDialogBaseProps {
...
 }
 
 export default function ViewVersionDialog(props: ViewVersionDialogProps) {
-  const { open, onClose, onDismiss, rightActions } = props;
+  return (
+    <Dialog
+      open={props.open}
+      onClose={props.onClose}
 
-  const resource = useStateResource<Resource, ViewVersionDialogProps>(props, {
```
Comment (rart) [inline ui/app/src/modules/Content/History/ViewVersionDialog.tsx:143]: Is there a pointless empty line here? remove if so.

File: ui/app/src/modules/Content/History/HistoryDialog.tsx [hunk 30]
```diff
@@ -240,70 +264,60 @@ export default function HistoryDialog(props: HistoryDialogProps) {
...
-              onClose: resetVersionsState()
-            }),
+            onClick: showHistoryDialog(),
             'aria-label': formatMessage(translations.backToHistoryList)
           }
         ]
       })
     );
+    dispatch(fetchContentTypes());
+    dispatch(fetchContentVersion({ path, versionNumber: version.versionNumber }));
```
Comment (rart) [inline ui/app/src/modules/Content/History/HistoryDialog.tsx]: batch actions

File: ui/app/src/modules/Content/Publish/PublishDialog.tsx [hunk 31]
```diff
@@ -405,21 +405,32 @@ function PublishDialogUI(props: PublishDialogUIProps) {
...
+    </>
   );
 }
 
-function PublishDialog(props: PublishDialogProps) {
+export default function PublishDialog(props: PublishDialogProps) {
+  return (
+    <Dialog
+      open={props.open}
+      onClose={props.onClose}
```
Comment (rart) [inline ui/app/src/modules/Content/Publish/PublishDialog.tsx]: x

File: ui/app/src/state/epics/dialogs.ts [hunk 32]
```diff
@@ -85,7 +85,7 @@ export default [
         // don't include the "CLOSE_*_DIALOG" action to avoid said loop.
         const onClose = getDialogState(type, state)?.onClose;
         return [
-          createClosedAction(type),
+          //createClosedAction(type),
```
Comment (rart) [inline ui/app/src/state/epics/dialogs.ts]: x

File: ui/app/src/state/reducers/dialogs/confirm.ts [hunk 33]
```diff
@@ -22,17 +22,13 @@ export default createReducer<GlobalState['dialogs']['confirm']>(
   { open: false },
   {
     [showConfirmDialog.type]: (state, { payload }) => ({
-      // By default, if no callback is specified, assume an ok button.
-      // To not have a "Ok" button, action creator must be called with onOk: null.
-      // This allows easily sending information dialogs with a ok button.
```
Comment (rart) [inline ui/app/src/state/reducers/dialogs/confirm.ts:27]: why remove comment? this is still the case. put back?

File: ui/app/src/state/reducers/dialogs/confirm.ts [hunk 34]
```diff
@@ -22,17 +22,13 @@ export default createReducer<GlobalState['dialogs']['confirm']>(
...
+      onClose: closeConfirmDialog(),
+      onClosed: confirmDialogClosed(),
       onOk: closeConfirmDialog(),
       ...payload,
       open: true
     }),
-    // Leaving/carrying the onClose as otherwise, it would never get executed since,
-    // the callback would be gone by the time the dialog animation is done (react would have
-    // already updated the props. Every time the dialog is opened, the onClose is cleansed anyway.
-    [closeConfirmDialog.type]: (state) => ({ open: false, onClose: state.onClose }),
```
Comment (rart) [inline ui/app/src/state/reducers/dialogs/confirm.ts:35]: removing the onclose will cause it not executing

File: ui/app/src/state/reducers/dialogs/delete.ts [hunk 35]
```diff
@@ -22,12 +22,14 @@ export default createReducer<GlobalState['dialogs']['delete']>(
...
     [showDeleteDialog.type]: (state, { payload }) => ({
+      ...state,
       onClose: closeDeleteDialog(),
+      onClosed: deleteDialogClosed(),
       onDismiss: closeDeleteDialog(),
       ...payload,
       open: true
     }),
-    [closeDeleteDialog.type]: (state) => ({ open: false, onClose: state.onClose }),
+    [closeDeleteDialog.type]: (state) => ({ ...state, open: false }),
```
Comment (rart) [inline ui/app/src/state/reducers/dialogs/delete.ts:32]: same deal. looks like you changed all. we might have lost support for onclose with this.

File: ui/app/src/utils/hooks.ts [hunk 36]
```diff
@@ -248,3 +245,10 @@ export function useMinimizeDialog(initialTab: MinimizedDialog) {
 
   return state?.minimized ?? initialTab.minimized;
 }
+export const useDialogOnClosed = (onClosed) => {
```
Comment (rart) [inline ui/app/src/utils/hooks.ts]: rename to useOnUnmount

File: ui/app/src/utils/hooks.ts [hunk 37]
```diff
@@ -248,3 +245,10 @@ export function useMinimizeDialog(initialTab: MinimizedDialog) {
...
+  useOnMount(() => {
```
Comment (rart) [inline ui/app/src/utils/hooks.ts]: Use useeffect directly to avoid the unnecessary abstraction

**Review by rart — CHANGES_REQUESTED** [review]
Please address comments

File: ui/app/src/components/Dialogs/ConfirmDialog.tsx [hunk 38]
```diff
@@ -24,6 +23,8 @@ import Button from '@material-ui/core/Button';
 import React, { PropsWithChildren } from 'react';
 import { defineMessages, useIntl } from 'react-intl';
 import StandardAction from '../../models/StandardAction';
+import { Dialog } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/components/Dialogs/ConfirmDialog.tsx]: Still incorrect. Import from their individual files.

File: ui/app/src/components/SystemStatus/ErrorDialog.tsx [hunk 39]
```diff
@@ -14,14 +14,14 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import Dialog from '@material-ui/core/Dialog';
 import React, { PropsWithChildren } from 'react';
 import IconButton from '@material-ui/core/IconButton';
 import CloseIcon from '@material-ui/icons/CloseRounded';
-import { createStyles, makeStyles, Theme } from '@material-ui/core';
+import { createStyles, Dialog, makeStyles, Theme } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/components/SystemStatus/ErrorDialog.tsx]: Incorrect. Import from their individual files.

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx (hunk 26 above)
Comment (rart) [reply ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: Still pending 👆🏻

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 40]
```diff
@@ -243,17 +252,17 @@ export default function DeleteDialog(props: DeleteDialogProps) {
 
   useEffect(() => {
     open &&
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: We no longer need this, right? 👆🏻

File: ui/app/src/modules/Content/Dependencies/DependenciesDialog.tsx [hunk 41]
```diff
@@ -534,41 +502,57 @@ function DependenciesDialogUI(props: DependenciesDialogUIProps) {
...
 };
 
-function DependenciesDialog(props: DependenciesDialogProps) {
-  const { open, item, dependenciesShown, onClose, onDismiss } = props;
+export default function DependenciesDialog(props: DependenciesDialogProps) {
+  return (
+    <Dialog
+      open={props.open}
+      onClose={props.onClose}
+      fullWidth={true}
```
Comment (rart) [inline ui/app/src/modules/Content/Dependencies/DependenciesDialog.tsx]: Use shortcut syntax

File: ui/app/src/modules/Content/History/CompareVersionsDialog.tsx [hunk 42]
```diff
@@ -15,40 +15,43 @@
...
+import { Resource } from '../../../models/Resource';
+import ListItemText from '@material-ui/core/ListItemText';
 import ExpansionPanel from '@material-ui/core/ExpansionPanel';
 import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
-import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
-import Typography from '@material-ui/core/Typography';
 import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMoreRounded';
+import SingleItemSelector from '../Authoring/SingleItemSelector';
+import { Dialog } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/modules/Content/History/CompareVersionsDialog.tsx]: Still pending 👆🏻

File: ui/app/src/modules/Content/Dependencies/DependenciesDialog.tsx [hunk 43]
```diff
@@ -55,6 +52,8 @@ import Checkbox from '@material-ui/core/Checkbox';
 import Radio from '@material-ui/core/Radio';
 import EmbeddedLegacyEditors from '../../Preview/EmbeddedLegacyEditors';
 import { ApiResponse } from '../../../models/ApiResponse';
+import SingleItemSelector from '../Authoring/SingleItemSelector';
+import { Dialog } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/modules/Content/Dependencies/DependenciesDialog.tsx]: Still pending 👆🏻

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 44]
```diff
@@ -32,26 +33,26 @@ import createStyles from '@material-ui/core/styles/createStyles';
...
 import { Resource } from '../../../models/Resource';
 import TextField from '@material-ui/core/TextField';
-import Dialog from '@material-ui/core/Dialog';
 import DialogHeader from '../../../components/Dialogs/DialogHeader';
 import DialogBody from '../../../components/Dialogs/DialogBody';
 import { SuspenseWithEmptyState } from '../../../components/SystemStatus/Suspencified';
 import DialogFooter from '../../../components/Dialogs/DialogFooter';
 import Button from '@material-ui/core/Button';
 import CircularProgress from '@material-ui/core/CircularProgress';
+import { Dialog } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: Still pending 👆🏻

File: ui/app/src/modules/Content/Authoring/NewContentDialog.tsx (hunk 24 above)
Comment (rart) [reply ui/app/src/modules/Content/Authoring/NewContentDialog.tsx]: Still pending 👆🏻

File: ui/app/src/modules/Content/History/HistoryDialog.tsx [hunk 45]
```diff
@@ -47,6 +46,9 @@ import {
   showHistoryDialog,
   showViewVersionDialog
 } from '../../../state/actions/dialogs';
+import SingleItemSelector from '../Authoring/SingleItemSelector';
+import { Dialog } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/modules/Content/History/HistoryDialog.tsx]: Still pending 👆🏻

File: ui/app/src/modules/Content/History/HistoryDialog.tsx (hunk 30 above)
Comment (rart) [reply ui/app/src/modules/Content/History/HistoryDialog.tsx]: Still pending 👆🏻

**Review by rart — CHANGES_REQUESTED** [review]
Missed a few comments from last review. **Make sure** you click on "Load hidden" — from _this_ and _previous_ review — to resolve all.
image

**Comment by rart** [comment]
Please solve conflicts

File: ui/app/src/models/Version.ts (hunk 23 above)
Comment (aachavarria) [reply ui/app/src/models/Version.ts:17]: yes

**Review by rart — APPROVED** [review]

**Pull request merged**